pynicotine/gtkgui/userbrowse.py

```diff
@@ -687,24 +789,233 @@ def on_select_dir(self, selection):
 
         self.set_directory(iterator.user_data)
 
-    def on_expand_row_accelerator(self, *args):
-        """ Right: expand row """
+    def on_select_file(self, selection):
+
+        if selection is None:
+            return
+
+        self.num_selected_files = selection.count_selected_rows()
+
+    """ Key Bindings (FolderTreeView) """
+
+    def on_folder_collapse_accelerator(self, *args):
+        """ Left: collapse row
+            Shift+Left (Gtk) | "-" | "/" (Gtk) | """
 
         path, _focus_column = self.FolderTreeView.get_cursor()
+
+        if path is None:
+            return False
+
+        self.FolderTreeView.collapse_row(path)
+        return True
+
+    def on_folder_expand_accelerator(self, *args):
+        """ Right: expand row
+            Shift+Right (Gtk) | "+" (Gtk) |    """
+
+        path, _focus_column = self.FolderTreeView.get_cursor()
+
+        expandable = self.FolderTreeView.expand_row(path, False)
+
+        if not expandable and len(self.file_store) > 0:
+            self.FileTreeView.grab_focus()
+
+        return True
+
+    def on_folder_collapse_sub_accelerator(self, *args):
+        """ \backslash: collapse or expand to show subs """
+
+        path, _focus_column = self.FolderTreeView.get_cursor()
+
+        self.FolderTreeView.collapse_row(path)  # show 2nd level
         self.FolderTreeView.expand_row(path, False)
+
         return True
 
-    def on_collapse_row_accelerator(self, *args):
-        """ Left: expand row """
+    def on_folder_expand_sub_accelerator(self, *args):
+        """ =equal: expand only (dont move focus)   """
 
         path, _focus_column = self.FolderTreeView.get_cursor()
-        self.FolderTreeView.collapse_row(path)
+        self.FolderTreeView.expand_row(path, False)
+        return True
+
+    def on_folder_focus_filetree_accelerator(self, *args):
+        """ Shift+Enter: focus selection over FileTree  """
+
+        if len(self.file_store) >= 1:
+            self.FileTreeView.grab_focus()
+
+        else:
+            self.on_folder_expand_sub_accelerator()
+            # ToDo: Expand all branch sub-nodes (*)                                                             # ToDo #
```

Comment:
This would be nice but isn't essential. If it can't be easily done then I will remove this comment line.